libproject/devstart.rb

```diff
@@ -0,0 +1,155 @@
+# UI project management commands and command-line flag definitions.
+
+require "optparse"
+require "set"
+require_relative "../utils/workbench"
+
+def ensure_docker(cmd_name, args)
+  unless Workbench::in_docker?
+    exec *(%W{docker-compose run --rm jupyterlab-extensions ./project.rb #{cmd_name}} + args)
+  end
+end
+
+# Creates a default command-line argument parser.
+# command_name: For help text.
+def create_parser(command_name)
+  OptionParser.new do |parser|
+    parser.banner = "Usage: ./project.rb #{command_name} [options]"
+    parser
+  end
+end
+
+def install_dependencies()
+  common = Common.new
+  common.run_inline %W{docker-compose run --rm jupyterlab-extensions yarn install}
+end
+
+Common.register_command({
+  :invocation => "install-dependencies",
+  :description => "Installs dependencies via yarn.",
+  :fn => Proc.new { |*args| install_dependencies(*args) }
+})
+
+def swagger_regen(cmd_name)
+  ensure_docker cmd_name, []
+
+  common = Common.new
+  common.run_inline %W{yarn run codegen}
+end
+
+Common.register_command({
+  :invocation => "swagger-regen",
+  :description => "Regenerates API client libraries from Swagger definitions.",
+  :fn => Proc.new { |*args| swagger_regen("swagger-regen") }
+})
+
+class BuildOptions
+  # Keep in sync with .angular-cli.json.
```

Comment:
SHould drop this logic as it's not used currently (separately, we'll need to figure out how to load the API url from within the extension). Could be we compile that in, not sure.